src/mesh.cpp

```diff
@@ -102,6 +121,70 @@ StructuredMesh::bin_label(int bin) const {
   }
 }
 
+//==============================================================================
+// Unstructured Mesh implementation
+//==============================================================================
+
+UnstructuredMesh::UnstructuredMesh(pugi::xml_node node) : Mesh(node) {
+  n_dimension_ = 3;
+
+  // check the mesh type
+  if (check_for_node(node, "type")) {
+    auto temp = get_node_value(node, "type", true, true);
+    if (temp != "unstructured") {
+      fatal_error(fmt::format("Invalid mesh type: {}", temp));
+    }
+  }
+
+  // get the filename of the unstructured mesh to load
+  if (check_for_node(node, "filename")) {
+    filename_ = get_node_value(node, "filename");
+  }
```

Comment:
```suggestion
    if (!file_exists(filename_)) {
      fatal_error("Mesh file '" + filename_ + "' does not exist!");
    }
  }
```

Reply:
Not sure how to do a similar commit suggestion for a line that hasn't changed... anyways, if you like this, you'd need to `#include "openmc/file_utils.h"` as well.

This check isn't strictly necessary since it still fails, but without it, you get a big dump of a stack trace that is kind of messy and just says "Error reading ExodusII file", not necessarily the more specific message that it's missing.

Reply:
Mmm yeah this is much better than an Exodus error. I'll include it and the necessary header.